lib/puppet/provider/datacat_collector/datacat_collector.rb

```diff
@@ -9,6 +9,16 @@ def exists?
       r.is_a?(Puppet::Type.type(:datacat_fragment)) && ((our_names & [ r[:target] ].flatten).size > 0)
     end
 
+    # decrypt any encrypted fragments
+    if defined?(Puppet_X::Binford2k::NodeEncrypt)
+      fragments.each do |fragment|
+        fragment[:data].each do |key,value|
+          next unless Puppet_X::Binford2k::NodeEncrypt.encrypted?(value)
```

Comment:
maybe we could do both approaches, flag the fragment with a parameter, then iterate each value like I'm currently doing. Thoughts?